buildSrc/src/main/groovy/org/elasticsearch/gradle/VersionCollection.groovy

```diff
@@ -140,7 +140,11 @@ class VersionCollection {
                     }
                     // caveat 0 - now dip back 2 versions to get the last supported snapshot version of the line
                     Version highestMinor = getHighestPreviousMinor(currentVersion.major - 1)
-                    maintenanceBugfixSnapshot = replaceAsSnapshot(highestMinor)
+                    if (highestMinor == null) {
```

Comment:
is this a pre-existing bug?

Reply:
Yes. Though `maintenanceBugfixSnapshot` should always be null in this case so I pushed https://github.com/elastic/elasticsearch/pull/32937/commits/40d70b40fa254c184f5fa3b6aa440b6e8f226d87